simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,400 @@
+        """
+        people = []
+        for i in range(self.numberOfPeople):
+            person = Person()
+
+            person.index = i
+
+            truncnormGenerator = getTruncatedNormal(self.placesLivedMean, self.placesLivedSD, 
+                                                    1, self.placesLivedUpper)
+            numberOfPlacesLived = int(truncnormGenerator.rvs())
+            person.lived = [0] * len(self.places)
+            placesLived = np.random.choice(self.places, numberOfPlacesLived, replace = False)
+            # One hot encoding of places lived by a person.
+            person.lived = [1 if i in placesLived else 0 for i in range(len(self.places))]
+
+            person.globalLikes = dict(zip(self.globalThings, 
+                                            np.random.uniform(0, 1, len(self.globalThings))))
+
+            person.localLikes = self.__getLikeability(person)
+            
+            people.append(person)
+
+        return people
+
+    def __generateKnows(self, people):
+        """
+        Return the knows truth matrix for all people in the dataset.
+        """
+        knowsMatrix = np.zeros((self.numberOfPeople, self.numberOfPeople))
+        for pair in combinations(people, 2):
+            similarity = self.__calculateSimilarity(pair[0], pair[1])
+            knows = self.__flipCoin(similarity)
+
+            # Set both (i,j) and (j,i) since P1 knows P2 => P2 knows P1.
+            knowsMatrix[pair[0].index][pair[1].index] = knows
+            knowsMatrix[pair[1].index][pair[0].index] = knows
+
+        return knowsMatrix
+
+    def __writeDataToFile(self, data, paths):
+        """
+        Write data to file.
+        """
+        if self.outputDir:
+            if self.outputDir[0] == '~':
+                prefix = os.path.expanduser(self.outputDir)
+            else:
+                prefix = self.outputDir
+        else:
+            prefix = ''
+
+        for key in data:
+            # We have the same keys for data and its corresponding path.
+            with open(os.path.join(prefix, paths[key]), 'w') as f:
+                if key == 'config':
+                    json.dump(data[key], f, indent = 4, sort_keys = True)
+                else:
+                    for dataItem in data[key]:
+                        f.write("\t".join([str(d) for d in dataItem]) + "\n")
+
+    def __prepareDataToWrite(self, knowsMatrix, people):
+        """
+        Prepare data for writing to file.
+        """
+        n = knowsMatrix.shape[1]
+
+        knowsData = []
+        knowsObs = []
+        knowsTarget = []
+        knowsTruth = []
+        
+        totalCombinations = knowsMatrix.shape[0] * (knowsMatrix.shape[1] - 1)
+        targets = np.random.choice(range(totalCombinations), 
+                                    int(self.targetSplit * totalCombinations), replace = False)
+        for i in range(knowsMatrix.shape[0]):
+            for j in range(knowsMatrix.shape[1]):
+                if i != j:
```

Comment:
You have 4 checks for `i != j`, literally half the lines in this block check!
Instead, just continue if `i == j` at the beginning of this loop.

Reply:
Noted.